### PR TITLE
test-autoscaling stop using deprecated framework.ExpectNotEqual

### PR DESCRIPTION
**Pull request by wlq1212**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
test-autoscaling stop using deprecated framework.ExpectNotEqual
#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/pull/115961
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 1]
```diff
@@ -121,7 +121,7 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
 		framework.ExpectNoError(err)
 		nodeCount = len(nodes.Items)
 		ginkgo.By(fmt.Sprintf("Initial number of schedulable nodes: %v", nodeCount))
-		framework.ExpectNotEqual(nodeCount, 0)
+		gomega.Expect(nodeCount).ToNot(gomega.BeEmpty())
```
Comment (carlory) [inline test/e2e/autoscaling/cluster_size_autoscaling.go:124]:
```suggestion
		gomega.Expect(nodes.Items).ToNot(gomega.BeEmpty())
```

**Review by pohly — APPROVED** [review]
/approve

**Pull request merged**